### PR TITLE
typecheck: Add support for augmented arithmetic assignment

### PR DESCRIPTION
**Pull request by simeonkr**

### CODE REVIEW

**Comment by simeonkr** [comment]
The failing test (`test_assign.test_augassign`) should pass once #421 has been merged.

File: python_ta/transforms/type_inference_visitor.py [hunk 1]
```diff
@@ -217,6 +217,32 @@ def visit_assign(self, node: astroid.Assign) -> None:
             if isinstance(type_result, TypeFail):
                 node.inf_type = type_result
 
+    def visit_augassign(self, node: astroid.AugAssign) -> None:
+        if isinstance(node.value, astroid.Name):
```
Comment (david-yz-liu) [inline python_ta/transforms/type_inference_visitor.py]: I'm not sure why this is a special case; seems like it's okay to use the `.inf_type` here as long as `self._assign_type` is called below?

File: python_ta/transforms/type_inference_visitor.py [hunk 2]
```diff
@@ -217,6 +217,32 @@ def visit_assign(self, node: astroid.Assign) -> None:
...
+
+        # lookup method for augmented arithmetic assignment
+        method_name = BINOP_TO_METHOD[node.op]
+        binop_result = self._handle_call(node, method_name, right_inf_type, node.value.inf_type)
+        if isinstance(binop_result, TypeFail):
+            # on failure, fallback to method corresponding to standard operator
+            method_name = BINOP_TO_METHOD[INPLACE_TO_BINOP[node.op]]
+            alt_binop_result = self._handle_call(node, method_name, right_inf_type, node.value.inf_type,
+                                               error_func=binop_error_message)
+            if isinstance(alt_binop_result, TypeInfo):
```
Comment (david-yz-liu) [inline python_ta/transforms/type_inference_visitor.py]: This check shouldn't be necessary, since `self._assign_type` accepts failables.

**Comment by david-yz-liu** [comment]
@simeonkr I'll merge this in after the conflicts have been resolved.

**Comment by simeonkr** [comment]
There's an actual failing test here, possibly introduced by the recently merged PRs. I'll fix it right away.

**Comment by david-yz-liu** [comment]
Did @justinpark715 fix the test? I can't remember. You should rebase this one anyway now.

**Comment by simeonkr** [comment]
The tests were incorrect anyways and failed because they require support for the sort of attribute-based inference that we discussed on Friday. I've added correct tests for those two cases, but they're skipped for now.

**Pull request merged**